framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java

```diff
@@ -0,0 +1,70 @@
+package org.motechproject.uitest.page;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+/**
+ * A class that represents data services page. Has methods which check functionality of
+ * data browser, schema editor and data services settings.
+ */
+
+public class DataServicesPage extends AbstractBasePage {
+
+    public static final By ENTITY_NAME_FIELD = By.name("inputEntityName");
+    public static final By NEW_ENTITY_BUTTON = By.id("newEntityButton");
+    public static final By SAVE_ENTITY_BUTTON = By.id("saveNewEntityButton");
+    public static final By DATA_SERVICES_BUTTON = By.id("modulelink_data-services");
+    public static final By SCHEMA_EDITOR_BUTTON = By.id("mdsTab_schemaEditor");
+    public static final By BROWSE_INSTANCES_BUTTON = By.id("browseInstancesButton");
+    public static final By ADD_NEW_INSTANCE_BUTTON = By.id("addNewInstanceButton");
+    public static final By ENTITY_SPAN = By.id("select2-chosen-2");
+
+    public static final String HOME_PATH = "/module/server/home#";
+
+    public DataServicesPage(WebDriver driver) {
+        super(driver);
+    }
+
+    /**
+     * Method creates new entity in MDS schema editor.
+     * @param entityName new entity name
+     * @return method returns text that appears in schema editor entity input after creating new entity, should be the same as new entity name, should be checked in tests
+     */
+    public String createNewEntity(String entityName) throws InterruptedException {
+        waitForElement(DATA_SERVICES_BUTTON);
+        clickWhenVisible(DATA_SERVICES_BUTTON);
+        waitForElement(SCHEMA_EDITOR_BUTTON);
```

Comment:
For example, you shouldn't have to wait before doing clickWhebVisible, since that method is doing the waiting